pkg/datapath/iptables/iptables.go

```diff
@@ -587,36 +607,42 @@ func (m *IptablesManager) InstallRules(ifName string) error {
 				return err
 			}
 
-			// Masquerade all traffic from the host into the ifName
-			// interface if the source is not the internal IP
-			//
-			// The following conditions must be met:
-			// * Must be targeted for the ifName interface
-			// * Must be targeted to an IP that is not local
-			// * Tunnel mode:
-			//   * May not already be originating from the masquerade IP
-			// * Non-tunnel mode:
-			//   * May not orignate from any IP inside of the cluster range
-			if err := runProg("iptables", []string{
-				"-t", "nat",
-				"-A", ciliumPostNatChain,
-				"!", "-s", ingressSnatSrcAddrExclusion,
-				"!", "-d", node.GetIPv4AllocRange().String(),
-				"-m", "comment", "--comment", "cilium host->cluster masquerade",
-				"-j", "SNAT", "--to-source", node.GetHostMasqueradeIPv4().String()}, false); err != nil {
-				return err
+			if option.Config.Tunnel != option.TunnelDisabled {
```

Comment:
So we didn't need the masquerade rule for host->cluster in the direct routing mode, and this removes it?

Or is this now covered by something like the `cilium hostport cluster masquerade` rule?

Reply:
That SNAT rule is required to ensure that reply traffic is sent back via the tunnel if we reach a remote node via the tunnel from a local source. If the source address selection picked a routable IP, the response may then be routed back directly if not SNATed.

Reply:
The change could be done in a separate PR as well. It's needed for ENI to avoid having a rule pointing to `cilium_host` which is meaningless in ENI as well as to node depend on the allocation range.